vnext/Desktop/DevSupportManager.cpp

```diff
@@ -89,21 +95,27 @@ void DevSupportManager::StartPollingLiveReload(const string &debugHost, std::fun
       try {
         Url url(DevServerHelper::get_OnChangeEndpointUrl(debugHost));
         auto const resolveResult = m_resolver.resolve(url.host, url.port);
-        tcp::socket socket{m_context};
-        connect(socket, resolveResult);
 
-        request<string_body> request{verb::get, url.Target(), 11 /*HTTP 1.1*/};
-        request.set(field::host, url.host);
-        request.set(field::user_agent, BOOST_BEAST_VERSION_STRING);
-        request.set(field::connection, "keep-alive");
+        tcp_stream stream{m_context};
+        stream.connect(resolveResult);
+
+        http::request<http::string_body> request{http::verb::get, url.Target(), 11 /*HTTP 1.1*/};
+        request.set(http::field::host, url.host);
+        request.set(http::field::user_agent, BOOST_BEAST_VERSION_STRING);
+        request.set(http::field::connection, "keep-alive");
 
-        write(socket, request);
+        http::write(stream, request);
         flat_buffer buffer;
-        response<dynamic_body> response;
-        read(socket, buffer, response);
+        http::response<http::dynamic_body> response;
+        http::read(stream, buffer, response);
 
         if (response.result_int() == 205 /*ResetContent*/ && !token.is_canceled())
           onChangeCallback();
+
+        error_code ec;
+        stream.socket().shutdown(tcp::socket::shutdown_both, ec);
+        if (ec && ec != errc::not_connected)
+          throw system_error(ec);
       } catch (const std::exception & /*e*/) {
```

Comment:
>exception [](start = 26, length = 9)

So this is catching all system_error right? Because system_error derives from std:exception. Should it be scoped to system_error? Or we expect other types of exceptions? If so this should be catch-all catch(...)

Reply:
Out of scope.